app/components/works/agreement_component.html.erb

```diff
@@ -2,11 +2,16 @@
   <header>Terms of Deposit *</header>
   <div class="row">
     <div class="form-check col-auto">
-      <%= form.check_box :agree_to_terms, required: true, class: 'form-check-input' %>
-      <%= form.label :agree_to_terms,
-       "I agree to the #{link_to 'SDR Terms of Deposit', '#termsOfDepositModal', data: { bs_toggle: 'modal', bs_target: '#termsOfDepositModal'}}".html_safe,
-       class: 'form-check-label' %>
-       <div class="invalid-feedback">You must agree to the Terms of Deposit.</div>
+      <% if form.object.agree_to_terms %>
+        You agreed to the <%= link_to 'SDR Terms of Deposit', '#termsOfDepositModal', data: { bs_toggle: 'modal', bs_target: '#termsOfDepositModal'} %>
+        <%= time_ago_in_words form.object.model.fetch(:work).depositor.last_work_terms_agreement %> ago.
+      <% else %>
+        <%= form.check_box :agree_to_terms, required: true, class: 'form-check-input' %>
+        <%= form.label :agree_to_terms,
+        "I agree to the #{link_to 'SDR Terms of Deposit', '#termsOfDepositModal', data: { bs_toggle: 'modal', bs_target: '#termsOfDepositModal'}}".html_safe,
+        class: 'form-check-label' %>
+        <div class="invalid-feedback">You must agree to the Terms of Deposit.</div>
+      <% end %>
     </div>
```

Comment:
Selectively show checkbox based on if the user has previously accepted the agreements in the specified timeframe.